### PR TITLE
bmi055: fixes for on-chip filter

### PR DESCRIPTION
**Pull request by bkueng**
While going through the BMI055 I noticed the on-chip filters are not configured as expected.
This changes:
- Accel: use cutoff of 62.5 Hz instead of 500 Hz
- Gyro: the cutoff frequency is coupled with the ODR and is fixed to 116 Hz
  at 1 kHz readout rate. So this PR does not change anything for the gyro.

Relevant parts from the datasheet:
screenshot_20190211_150946

screenshot_20190211_150929

@bresch can you test this on one of the vehicles?

### CODE REVIEW

**Comment by dagar** [comment]
To be consistent with the other IMU drivers you should probably add the PX4 side low pass as well.

 - IMU_ACCEL_CUTOFF
 - IMU_GYRO_CUTOFF

I want to get this out of the actual drivers, but it's what we currently have.

**Comment by dagar** [comment]
@bresch if you happen to still have a flight log from flying the BMI055 as primary with no filtering it might be interesting to compare. https://github.com/PX4/Firmware/pull/11432

File: src/drivers/imu/bmi055/BMI055_accel.cpp [hunk 1]
```diff
@@ -214,48 +214,34 @@ BMI055_accel::probe()
...
 		_accel_sample_rate = 2000;
 
 	} else {
 		return -EINVAL;
 	}
 
-	/* Write accel ODR */
-	modify_reg(BMI055_ACC_BW, clearbits, setbits);
-
 	return OK;
```
Comment (davids5) [inline src/drivers/imu/bmi055/BMI055_accel.cpp]: @bkueng - with the removal of the setting of the register does this all become a NOP?

image

**Comment by bkueng** [comment]
> To be consistent with the other IMU drivers you should probably add the PX4 side low pass as well.

Right that is the other part that was missing. I added them as well.

> @bkueng - with the removal of the setting of the register does this all become a NOP?

Nice catch. I assumed it was used elsewhere. I removed it completely.

**Comment by dagar** [comment]
Anyone able to give this a quick flight test on a pixhawk 4 with the icm20689 disabled?

**Comment by dagar** [comment]
@PX4/testflights could you try this on a pixhawk 4 or 4 mini? Start with any normal flight as a baseline. Then repeat, but this time manually disable the primary IMU (try `mpu6000 stop`). If that doesn't work it can be done via parameter (CAL_ACC0_EN,  CAL_GYRO0_EN).

**Comment by dannyfpv** [comment]
tested on pixhawk4 v5  normal flight
https://review.px4.io/plot_app?log=3a3ec430-965b-4421-ba1d-8204dd654dc5
disable the primary IMU stops and go on mission mode
https://review.px4.io/plot_app?log=44cb4cd5-fcc9-4e24-84d0-51587ac701bc

tested on pixhawk4 mini v5  normal flight
https://review.px4.io/plot_app?log=12d30737-48d5-41ab-b15a-5ec107b98dcc
disable the primary IMU stops and go on mission mode
https://review.px4.io/plot_app?log=042f3964-4836-4dd8-8ec1-ad4ca20c00c1

**Comment by dagar** [comment]
The estimator really didn't like the data from the BMI055.

image

**Comment by davids5** [comment]
@dagar - what would you suggest?

**Comment by dagar** [comment]
This is still an improvement over the existing state in master, but the bmi055 will need a closer look if we want it to truly be a safe fallback.

**Pull request merged**